libcloud/compute/drivers/ecs.py

```diff
@@ -0,0 +1,1390 @@
+        'Stopped': NodeState.STOPPED
+    }
+
+    VOLUME_STATE_MAPPING = {
+        'In_use': StorageVolumeState.INUSE,
+        'Available': StorageVolumeState.AVAILABLE,
+        'Attaching': StorageVolumeState.ATTACHING,
+        'Detaching': StorageVolumeState.INUSE,
+        'Creating': StorageVolumeState.CREATING,
+        'ReIniting': StorageVolumeState.CREATING}
+
+    SNAPSHOT_STATE_MAPPING = {
+        'progressing': VolumeSnapshotState.CREATING,
+        'accomplished': VolumeSnapshotState.AVAILABLE,
+        'failed': VolumeSnapshotState.ERROR}
+
+    def list_nodes(self, ex_node_ids=[], ex_filters=None):
+        """
+        List all nodes.
+
+        @inherits: :class:`NodeDriver.create_node`
+
+        :keyword  ex_node_ids: a list of node's ids used to filter nodes.
+                               Only the nodes which's id in this list
+                               will be returned.
+        :type   ex_node_ids: ``list`` of ``str``
+        :keyword  ex_filters: node attribute and value pairs to filter nodes.
+                              Only the nodes which matchs all the pairs will
+                              be returned.
+                              If the filter attribute need a json array value,
+                              use ``list`` object, the driver will convert it.
+        :type   ex_filters: ``dict``
+        """
+
+        params = {'Action': 'DescribeInstances',
+                  'RegionId': self.region}
+
+        if ex_node_ids:
+            if isinstance(ex_node_ids, list):
+                params['InstanceIds'] = json.dumps(ex_node_ids)
+            else:
+                raise AttributeError('ex_node_ids should be a list of '
+                                     'node ids.')
+
+        if ex_filters:
+            if isinstance(ex_filters, dict):
+                params.update(ex_filters)
+            else:
+                raise AttributeError('ex_filters should be a dict of '
+                                     'node attributes.')
+
+        nodes = self._request_multiple_pages(self.path, params,
+                                             self._to_nodes)
+        return nodes
+
+    def list_sizes(self, location=None):
+        params = {'Action': 'DescribeInstanceTypes'}
+
+        resp_body = self.connection.request(self.path, params).object
+        size_elements = findall(resp_body, 'InstanceTypes/InstanceType',
+                                namespace=self.namespace)
+        sizes = [self._to_size(each) for each in size_elements]
+        return sizes
+
+    def list_locations(self):
+        params = {'Action': 'DescribeRegions'}
+
+        resp_body = self.connection.request(self.path, params).object
+        location_elements = findall(resp_body, 'Regions/Region',
+                                    namespace=self.namespace)
+        locations = [self._to_location(each) for each in location_elements]
+        return locations
+
+    def create_node(self, **kwargs):
```

Comment:
If possible, please also explicitly declare all the arguments here instead of using kwargs (`def create_node(self, name, size, image, ex_security_group_id, ...**kwargs`)).

I know some drivers still use `**kwargs`, but we want to avoid that going forward.